Source/WebKit/UIProcess/ios/DragDropInteractionState.h

```diff
@@ -92,8 +95,8 @@ class DragDropInteractionState {
     bool isPerformingDrop() const { return m_isPerformingDrop; }
     id<UIDragSession> dragSession() const { return m_dragSession.get(); }
     id<UIDropSession> dropSession() const { return m_dropSession.get(); }
-    BlockPtr<void()> takeDragStartCompletionBlock() { return WTFMove(m_dragStartCompletionBlock); }
-    BlockPtr<void(NSArray<UIDragItem *> *)> takeAddDragItemCompletionBlock() { return WTFMove(m_addDragItemCompletionBlock); }
+    BlockPtr<BOOL()> takeDragStartCompletionBlock() { return std::exchange(m_dragStartCompletionBlock, { }); }
+    BlockPtr<BOOL(NSArray<UIDragItem *> *)> takeAddDragItemCompletionBlock() { return std::exchange(m_addDragItemCompletionBlock, { }); }
```

Comment:
Is it not possible to express these as `BlockPtr<DragPreparationBlock>`?

Reply:
Hm...I don't think that works — the template arguments for `BlockPtr` are defined as such:

```
<R (Args...)>
```

...so there would need to be both a return value, followed by any number of other template arguments in parentheses. Using a simple ObjC++ test program:

```
#import "config.h"

#import <wtf/BlockPtr.h>

using BoolResultBlock = BOOL(^)(void);

int main(int argumentCount, const char* arguments[])
{
    BlockPtr<BoolResultBlock> wrapper = ^{
        return NO;
    };
    return 0;
}
```

...I get:

```
main.mm:9:31: error: implicit instantiation of undefined template 'WTF::BlockPtr<bool (^)()>'
    9 |     BlockPtr<BoolResultBlock> wrapper = ^{
      |                               ^
/Applications/Xcode.app/…/usr/local/include/wtf/BlockPtr.h:60:26: note: template is declared here
   60 | template<typename> class BlockPtr;
      |                          ^
1 error generated.
```

Reply:
Ah, yeah, I see what we do...

```cpp
template<typename R, typename... Args>
class BlockPtr<R (Args...)> {
public:
    using BlockType = R (^)(Args...);
```